tests/nest_tests/third_factor_stdp_synapse_test.py

```diff
@@ -304,25 +225,23 @@ def run_synapse_test(self, neuron_model_name,
             ax[-2].set_ylabel(u"Δw")
 
             ax[-1].plot(t_hist, w_hist, marker="o")
-            if sim_ref:
-                ax[-1].plot(t_hist, w_hist_ref, linestyle="--", marker="x", label="ref")
             ax[-1].set_ylabel("w")
-
             ax[-1].set_xlabel("Time [ms]")
             for _ax in ax:
                 if not _ax == ax[-1]:
                     _ax.set_xticklabels([])
                 _ax.grid(which="major", axis="both")
                 _ax.xaxis.set_major_locator(matplotlib.ticker.FixedLocator(np.arange(0, np.ceil(sim_time))))
                 _ax.set_xlim(0., sim_time)
+
             fig.savefig("/tmp/stdp_third_factor_synapse_test" + fname_snip + ".png", dpi=300)
 
         # verify
         MAX_ABS_ERROR = 1E-6
-        idx = np.where(np.abs(third_factor_trace) < 1E-3)[0]  # find where third_factor_place is (almost) zero
+        idx = np.where(np.abs(third_factor_trace) < 1E-3)[0]  # find where third_factor_trace is (almost) zero
         times_dw_should_be_zero = timevec[idx]
-        for time_dw_should_be_zero in times_dw_should_be_zero:
+        for time_dw_should_be_zero in times_dw_should_be_zero[:-1]:
             _idx = np.argmin((time_dw_should_be_zero - np.array(t_hist))**2)
-            assert np.abs(np.diff(w_hist)[_idx]) < MAX_ABS_ERROR
+            assert np.abs(np.diff(w_hist)[_idx + 1]) < MAX_ABS_ERROR
```

Comment:
This test is failing for my PR as well. Why does this need to be changed? Is it because of a change in NEST upstream?

Reply:
I think it's a fluke, due to the fact that our test timing is randomly generated in that test and different for every run, coupled with an off-by-one bug on the actual check.

Reply:
But the test is consistently failing for the other PRs #950 and #954.